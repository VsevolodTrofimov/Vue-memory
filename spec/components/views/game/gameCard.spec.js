import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'

import gameCard from '@/src/components/views/Game/Card.vue'


const VSSR = createRenderer()
const localVue = createLocalVue()
localVue.config.productionTip = false


describe('Views', () => {
  describe('Game View Card', () => {
    it('matches snapshot', () => {
      const wrapper = mount(gameCard)

      VSSR.renderToString(wrapper.vm, (err, str) => {
        expect(str).toMatchSnapshot()
      })
    })

    it('matches hidden snapshot', () => {
      const wrapper = mount(gameCard, {
        propsData: {hidden: true}
      })

      VSSR.renderToString(wrapper.vm, (err, str) => {
        expect(str).toMatchSnapshot()
      })
    })

    it('matches flipped snapshot', () => {
      const wrapper = mount(gameCard, {
        propsData: {
          flipped: true,
          path: '/static/img.png'
        }
      })

      VSSR.renderToString(wrapper.vm, (err, str) => {
        expect(str).toMatchSnapshot()
      })
    })
  })
})
