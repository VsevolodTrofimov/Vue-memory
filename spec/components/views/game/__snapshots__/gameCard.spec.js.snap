// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views Game View Card matches flipped snapshot 1`] = `
<div appear="" name="card-flip" mode="out-in" class="card">
    <img src="/static/img.png" class="card__image">
</div>
`;

exports[`Views Game View Card matches hidden snapshot 1`] = `
<div appear="" name="card-flip" mode="out-in" class="card card--hidden">
    <img class="card__image">
</div>
`;

exports[`Views Game View Card matches snapshot 1`] = `
<div appear="" name="card-flip" mode="out-in" class="card card--back">
    <img src="/static/cards/back.png" class="card__image">
</div>
`;
